08-online-eval/README.md

```diff
@@ -0,0 +1,126 @@
+# Run LLM evaluations against data generated from a running app
+
+This exercise teaches you to evaluate LLM responses for relevancy and
+hallucinations using the LLM-as-a-judge pattern for data generated from
+a running app. You’ll test `OpenAIClient.chat()`, exporting traces to
+[Phoenix][phoenix] to then load and process with [Phoenix Evals][phoenix-evals].
+
+```mermaid
+sequenceDiagram
+    participant App as main.py
+    participant Client as OpenAIClient
+    participant Phoenix
+    participant Evals
+    participant LLM as Evaluation LLM
+
+    App ->> Client: Call chat(model, message)
+    activate Client
+    Client -->> App: Return reply
+    deactivate Client
+    App ->> Phoenix: Send reply, input, context in trace
+    activate Evals
+    Evals ->> Phoenix: Fetch spans without evals
+    activate Phoenix
+    Phoenix -->> Evals: Return spans
+    deactivate Phoenix
+    Evals ->> LLM: Request scores for metrics
+    activate LLM
+    LLM -->> Evals: Return scores
+    deactivate LLM
+    Evals -->> Phoenix: Upload scores
+    deactivate Evals
+```
+
+## Start Phoenix
+
+Phoenix is a trace backend optimized for Gen AI. Start it to allow collecting
+traces and eval results.
+
+```bash
+docker compose up -d phoenix
+```
+
+## Send traces
+
+Run the simple Gen AI application in [main.py](main.py). It will export telemetry
+to Phoenix. You can run it a few times to have a few sample traces to check out.
+
+<details>
+<summary>Docker</summary>
+
+```bash
+docker compose run --build --rm main
+```
+
+</details>
+
+<details>
+<summary>Shell</summary>
+
+
+Install dependencies:
+```bash
+pip install -r requirements.txt
+pip install -r requirements-dev.txt
+```
+
+Bootstrap instrumentation:
+```bash
+edot-bootstrap --action=install
+```
+
+Run the test:
+```bash
+OTEL_METRICS_EXPORTER=none OTEL_LOGS_EXPORTER=none python3 main.py
```

Comment:
Since Phoenix doesn't accept metrics or logs and python logs when they're not present

Reply:
and I suppose also that we don't want to taint ../.env cool

Reply:
@mikeldking maybe we should have fake endpoints for these? I thought I maybe raised an issue on it...

Reply:
> and I suppose also that we don't want to taint ../.env cool

Yeah. I consideered a second .env file in this directory for just these two but went with the inline approach instead